### PR TITLE
Fixes not being able to insert objects into closets, and being unable to throw

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
fix: Objects and items can now be inserted into closets, like before.
fix: Fixes a bug which prevented carbons from throwing items.
/:cl:

Essentially, the original way the insert() proc was structured, I
overlooked an else-if chain that was skipped if something was an item,
because I thought it only existed to type-check the item to access its
flags.

These are the changed lines from the NODROP PR. https://github.com/tgstation/tgstation/pull/42109/files#diff-2b130b854bf2e2a5e76320246520eea4L171

closes #42109

### CODE REVIEW

**Comment by Cyberboss** [comment]
Does this fix throwing?

**Comment by coiax** [comment]
No. Haven't found what's causing the throwing bug yet.

**Comment by coiax** [comment]
@Cyberboss Okay, it's ready.

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -169,7 +169,7 @@
 				if(start_T && end_T)
 					log_combat(src, throwable_mob, "thrown", addition="grab from tile in [AREACOORD(start_T)] towards tile at [AREACOORD(end_T)]")
 
-	else if((!I.item_flags & ABSTRACT) && !I.has_trait(TRAIT_NODROP))
+	else if(!CHECK_BITFIELD(I.item_flags, ABSTRACT) && !I.has_trait(TRAIT_NODROP))
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/carbon.dm]: why did this break, it should have worked fine before

Comment (SpaceManiac) [reply code/modules/mob/living/carbon/carbon.dm]: order of operations mistake: `(!I.item_flags & ABSTRACT)` should be `!(I.item_flags & ABSTRACT)`

Comment (optimumtact) [reply code/modules/mob/living/carbon/carbon.dm]: ahh

**Pull request merged**